cmd/kubeadm/app/phases/kubelet/flags.go

```diff
@@ -80,6 +82,9 @@ func buildKubeletArgMap(opts kubeletFlagsOpts) map[string]string {
 		} else {
 			kubeletFlags["cgroup-driver"] = driver
 		}
+		if opts.pauseImage != "" {
```

Comment:
Should this code take into account a value stored in `opts.nodeRegOpts.kubeletExtraArgs["pod-infra-container-image"]`?

Reply:
i need to revisit this PR, but its doing it the other way around at the moment.
whatever image repo the user sets in the config we pass that to the kubelet extra args to form the pod-infra-container-image URL.

Reply:
I guess my question was could/should it account for both methods? Seems simple enough.

If `pod-infra-container-image` were specified explicitly in `kubeletExtraArgs` it seems that should be chosen over the value calculated from the user-configured image repo.

Reply:
ok, understood.
i'm pretty sure that the value from kubeletExtraArgs (user value) would already override the value we set dynamically:
https://github.com/kubernetes/kubernetes/blob/0dd46c29ae9300f850882603536f08be846a184c/cmd/kubeadm/app/phases/kubelet/flags.go#L65